### DISCUSSION TITLE
Screen Auto-Brightness

### DISCUSSION THREAD

**Discussion by doobes** [elementary/triage#616] [Q&A]
Running v8.0, but this issue existed on v7.1 as well.

Dell XPS 13 9333 laptop. 

The screen Auto_Brightness daemon is constantly adjusting the brightness.

How do I turn it off completely?

Thanks

**Comment by doobes** [comment]
I've tried various solutions, to no avail.

Would someone please respond?

**Comment by doobes** [comment]
So, 

I've posted this at the Ubuntu site. They closed the convo at it's in reference to EOS.

Will someone respond?

**Comment by jeremypw** [comment]
In Settings open the "Power" plug. There you will find a toggle to turn off automatic adjustment of screen brightness.  I hope this works for you.

**Comment by doobes** [comment]
Yep,

Done that.

Also did the `gsettings set org.gnome.settings-daemon.plugins.power ambient-enabled false` thing.

Still occurring.

Thoughts?

**Comment by jeremypw** [comment]
In that case it is probably an upstream issue or possibly a faulty sensor.  (I assume the sensor is not obstructed :wink: ).  The actual dirty work is done by an upstream daemon (e.g. https://github.com/mikhail-m1/illuminanced) which may not be fully compatible with your hardware.  I do not have an ambient light sensor so I cannot investigate further but there should be a way of monitoring the raw ambient light signal being received from the sensor.  I assume the daemon averages the values over a reasonable period of time to smooth out rapid fluctuations.  If the screen brightness varies randomly even when the toggle is off the either the daemon is not honoring the setting (seems unlikely if it is the built-in one), it is producing wildy variable readings or maybe the backlight is faulty?  Does the screen brightness change as expected if ALS (ambient light sensor) toggle is on and you cover the sensor manually?

If you have the problem on elementary but not on e.g. Ubuntu then I am puzzled as afaict elementary only changes the gsetting.

**Comment by doobes** [comment]
Thanks,

IIRC it behaved this way when I had ZorinOS installed on this machine.

This tends to tell me the sensor has gone flaky.

Is there a way to disable the daemon?

Thanks

**Reply by jeremypw** [reply]
Some machines have a way of disabling it in hardware - see for example https://www.dell.com/support/kbdoc/en-uk/000127395/what-is-the-ambient-light-sensor-feature-on-a-portable-computer-and-how-do-i-adjust-the-settings-kb-article-301197

Consult the support for your device to check.  

You can prevent a daemon being started if you can find the correct config file but the details may be hardware specific so I cannot help further., sorry.  Some info how to go about this may be found at e.g. https://unix.stackexchange.com/questions/333256/prevent-starting-processes-automatically

**Comment by zeebok** [comment]
Check your BIOS/UEFI menu. Some laptops, such as the XPS line, have these set in the BIOS and it always seems to override whatever you set in the OS.

**Comment by NutriPhil** [comment]
I have the same problem here, with an older intel Macbook Pro 13. Constantly adjusting brightness. Problem not occuring under Ubuntu 24.10. So, in my case, its most likely not a broken sensor. Did not try the power setting workaround yet.

**Comment by doobes** [comment]
Well,

Looking through the BIOS/UEFI settings, there is no way to disable this.

We can close this discussion.  

Thanks
